gfx/drivers/d3d12.c

```diff
@@ -999,13 +997,9 @@ d3d12_gfx_init(const video_info_t* video, const input_driver_t** input, void** i
 
    font_driver_init_osd(d3d12, false, video->is_threaded, FONT_DRIVER_RENDER_D3D12_API);
 
-   if (settings->bools.video_shader_enable)
-   {
-      const char* ext = path_get_extension(retroarch_get_shader_preset());
-
-      if (ext && string_is_equal(ext, "slangp"))
-         d3d12_gfx_set_shader(d3d12, RARCH_SHADER_SLANG, retroarch_get_shader_preset());
-   }
+   const char *shader_preset   = retroarch_get_shader_preset();
```

Comment:
Same here. Variables need to be declared either at the top of a code block or function.